### PR TITLE
[WIP] feat: new GUI with all pymmcore-widgets

### PR DESCRIPTION
**Pull request by fdrgsp**
I know this might be a too big PR to merge/review and I can try to split it, however it would be great to have your opinion first @tlambert03 @ianhi! 

The PR shows a possible new updated GUI for `napari-micromanager`.

Compare to the previous versions, here I used `qtpy` `QToolBar`. Some `pymmcore-widgets` have been added directly, other can be opened through some `QPushButton` with icons. **Every** widget is a `QDockWidget` so users can dock them to `napari` if they want. MDA and Sample Explorer are also both `QDockWidget` and can be opened with two dedicated  `QPushButton`.

https://user-images.githubusercontent.com/70725613/202931891-06d2347d-c1e3-4484-b595-e5d52c0a60ae.mov

---
I've also added few nice features including:
 - the _scale_ to the layers we acquire so if they are z-stacks we can visualize them in 3D in napari.
 - if you right-click with the mouse on any image/stack acquired, you can move the xy and/or z stage to the position clicked or you can add the position to the MDA or Explorer.

https://user-images.githubusercontent.com/70725613/202946194-7eb8c28b-bda1-4192-8c69-0cbc36b3e2b6.mov

### CODE REVIEW

**Comment by tlambert03** [comment]
the screenshot is amazing 😍 

... but yeah, there's a bit too much going on in this PR.
I'd say it needs to be broken up into probably at least four PRs... (but maybe more?)

- the deletion of widgets in napari-micromanager in favor of pymmcore widgets should definitely come outside of a PR that adds new features like the toolbar.  You could consider even doing a separate PR for each widget swap/deletion (unless that would make for a lot of  merge conflicts)
- a pr to make things dock widgets
- a pr for the toolbar
- a pr for the click-to-move-stage feature

basically, anything that feels like a feature independent of other features should be it's own PR.  If one feature depends on another feature (e.g. if the stage feature needs a new widget) than it should still be a separate PR, but just note that it depends on another one.

**Comment by tlambert03** [comment]
but... just to reiterate.  it's really gorgeous.  looks far more useful and ergonomic.  and I can't wait for napari-micromanager to look like this :)

**Comment by tlambert03** [comment]
closed by #221
